### DISCUSSION TITLE
How to use the MutationListener

### DISCUSSION THREAD

**Discussion by emaborsa** [facebook/lexical#3911] [Q&A] [answered]
As already mentioned in another [post](https://github.com/facebook/lexical/issues/3073), there is the need of inlining the style instead class names. @trueadm said to use  the `MutationListener`, but I did not understand how to use it, can't find sample codes... except the [screenshot](https://user-images.githubusercontent.com/91624514/192383029-3b51c0bf-a293-409b-98e5-06fea1272ad6.png).
I am aware of the [documentation](https://lexical.dev/docs/concepts/listeners#registermutationlistener), but where is this code going to be added?

**Comment by acywatson** [comment]
Yea so you'd just set up a MutationListener for whatever node types you care about, then you'd get their associated DOM nodes using LexicalEditor.getElementByKey and apply styles directly to them using DOM APIs like element.style = whatever

**Reply by emaborsa** [reply]
Hi,
thanks, but where do I have to add the code? As I'understood, one MutationListener per Node type, but how can I add it if I am using nodes from `@lexical/*` packages?

**Reply by emaborsa** [reply]
I am still looking how to use mutationlistener.

**Reply by acywatson** [reply, marked as answer]
Add it in a plugin: https://lexical.dev/docs/react/plugins

**Reply by emaborsa** [reply]
My problem is not how to add it, but how to define it.

**Reply by acywatson** [reply]
Can you be specific about which part you don't understand?

https://lexical.dev/docs/concepts/listeners#registermutationlistener

**Reply by emaborsa** [reply]
Ok, now I got it. I didn't know where to put the code, did not understand that I had to define a simple plugin.
Thanks.
